### PR TITLE
doc: security: Add ETSI 303 645 standard

### PR DESCRIPTION
**Pull request by ceolin**
Add a new section for standards with ETSI 303-645 in the security related documentation

We can create individual prs for guidelines justification either for not applicable or a justification.

### CODE REVIEW

**Comment by ceolin** [comment]
@gregshue

**Comment by gregshue** [comment]
Nice! We really ought to include a link to download this freely available standard.  We also ought to include some definitions from the standard, because some of these are defined differently than many will assume (e.g. consumer IoT device, constrained device).

**Review by str4t0m — COMMENTED** [review]
Found one minor typo while skimming over the changes.

File: doc/security/standards/etsi-303645.rst [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+
+        .. _ETSI_Provision_6_1_4:
+    * - Provision 6.1-4
+      - If telemetry data is collected from consumer IoT devices and services, the
+        processing of personal data should be kept to the minimum necessary for the
+        intended functionality.
+      -
+
+        .. _ETSI_Provision_6_1_6:
+    * - Provision 6.1-6
```
[the remark this answers is not in the document]
Comment (ceolin) [reply doc/security/standards/etsi-303645.rst:424]: thanks !

**Review by nashif — CHANGES_REQUESTED** [review]
would be great if the context of why this is added, we are adding a standard, but how this relates to zephyr and what are we trying to do with those rules?

**Comment by gregshue** [comment]
> Nice! We really ought to include a link to download this freely available standard. We also ought to include some definitions from the standard, because some of these are defined differently than many will assume (e.g. consumer IoT device, constrained device).

Here's the link to the standard: https://www.etsi.org/deliver/etsi_en/303600_303699/303645/02.01.01_60/en_303645v020101p.pdf

and the companion conformance assessment to baseline requirements:

[https://www.etsi.org/deliver/etsi_ts/103700_103799/103701/01.01.01_60/ts_103701v010101p.pdf](https://www.etsi.org/deliver/etsi_ts/103700_103799/103701/01.01.01_60/ts_103701v010101p.pdf)

**Comment by dleach02** [comment]
@ceolin do you want to reopen this?

**Comment by ceolin** [comment]
@dleach02 Yes, I still want to have a conclusion about i. That is not about any sort of compliance but an assessment of what Zephyr does/provides.

**Comment by gregshue** [comment]
Ultimately this is about choosing which security standard(s) zephyr will be certified against.

**Comment by ceolin** [comment]
Some additional discussion
```
[7:58 AM] Ceolin, Flavio

Gregory A Shue one comment  / question about that secure development provision (and other in general actually), is Zephyr a component in the product BOM ? if yes, all these provisions apply the same way for it ? Asking it because I saw examples of products using Linux in that document, and Linux definitely does not follow  it

[7:59 AM] Ceolin, Flavio

So, that is very important thing to clear out ... what happens with third party components ? e.g we rely on mbedTLS for crypto, does mbedTLS need to follow these requirements as well ? 

[8:04 AM] Andrej Butok

A Secure Boot ROM + TFM should satisfy all needs.

[8:38 AM] Gregory A Shue

Ceolin, Flavio (External)

Gregory A Shue one comment  / question about that secure development provision (and other in general actually), is Zephyr a component in the product BOM ? if yes, all these provisions apply the same way for it ? Asking it because I saw examples of products using Linux in that document, and Linux de…


The Provisions apply to all the software embedded in the product. Provision 4-1 states "A justification shall be recorded for each recommendation in the present document that is considered to be not applicable or not fulfilled by the consumer IoT device." [etsi.org/deliver/etsi_tr/103600_103699/103621/01.01.01_60/tr_103621v010101p.pdf](https://www.etsi.org/deliver/etsi_tr/103600_103699/103621/01.01.01_60/tr_103621v010101p.pdf) section 8 provides guidance about handling recommendations. [etsi.org/deliver/etsi_ts/103700_103799/103701/01.01.01_60/ts_103701v010101p.pdf](https://www.etsi.org/deliver/etsi_ts/103700_103799/103701/01.01.01_60/ts_103701v010101p.pdf) is the Conformance Assessment standard that test houses must follow.


 

I don't know what justifications may be provided for Linux-based products. When the application is not running in the Linux kernel process manufacturers may be able to justify using a non-compliant Linux kernel.

[8:40 AM] Gregory A Shue

Ceolin, Flavio (External)

So, that is very important thing to clear out ... what happens with third party components ? e.g we rely on mbedTLS for crypto, does mbedTLS need to follow these requirements as well ?


TBD. What are acceptable justifications for using non-compliant third-party components?


[9:00 AM] Ceolin, Flavio

I don't know either ... but that is something that we need to understand

[9:20 AM] Gregory A Shue

Ceolin, Flavio (External)

I don't know either ... but that is something that we need to understand


AFAICT running Linux as a static monolithic kernel (no virtual addresses, no dynamic libraries) is closest Zephyr usages. Justifications that do not cover that are probably not relevant. Note that for Zephyr-based products, "applications" may include proprietary device drivers, code running in interrupt context, and even zero-latency interrupt handlers that preempt the OS critical sections.


[10:22 AM] Ceolin, Flavio

exactly, there is a huge chance on Zephyr that you are using third-party code for device drivers, different from Linux Zephyr does not have only native drivers ... But my point here is about these "processes" requirements / provisions 

[11:40 AM] Gregory A Shue

Manufacturers need to ship code that was developed under a process that meets the regulations for their markets. If a vendor is not going to provide drivers/etc. compliant with that process then the manufacturer may choose to go elsewhere, reverse-engineer the software, or do original engineering of a compliant solution. Which do you think a manufacturer prefers?

 

Zephyr has forks of several OSS projects. Zephyr Project could do the reverse engineering and provide verification suites in their forks. Zephyr Project could work with those OSS projects for them to provide it to ZP and other users. Zephyr Project could say usage of those modules is outside the current security scope and leave the problem to the chip vendors and/or manufacturers. I think Nordic wants to avoid the last one.

 

Frankly, TFM and MCUboot, and all HAL providers face the same problem as Zephyr Project. If these projects want their code to be in manufacturers products then they have to wrestle with the same market constraint.

[11:45 AM] Gregory A Shue

BTW, manufacturers consider Zephyr itself to be 3rd party code. 

[11:47 AM] David Brown

What kind of processes are you finding? For 303 645, it seems to be requiring things like revision control, which we would meet sufficiently.

[11:52 AM] Gregory A Shue

ETSI 303 645 Provision 5.6-9:

 

Provision 5.6-9 The manufacturer should follow secure development processes for software deployed on the device.

 

ETSI TS 103701 v1.1.1 Compliance Assessment for 5.6-9: (numbering added for this discussion)

The verdict PASS is assigned if the secure development covers:

(a) security training of developers; and 
(b) the requirement and design phases of the software; and 
(c) secure coding techniques; and 
(d) security tolling for the implementation phase; and 
(e) security testing; and 
(f) security reviews; and 
(g) archival of assets and information relevant to maintaining security of the software; and 
(h) secure deployment; and 
(i) if applicable, handling of third-party software providers; and 
 
(j) a confirmation for the implementation is given. 

The verdict FAIL is assigned otherwise.

 

[11:54 AM] Gregory A Shue

ETSI TR 103 621 v1.1.1 Guidance on Recommendations for Provision 5.6-9:

Provision 5.6-9 

"The manufacturer should follow secure development processes for software deployed on the device". (ETSI EN 303 645 [i.1]). 

 

EXAMPLE: The manufacturer is rebranding the device, which is obtained from a supplier under white label.

[11:56 AM] Ceolin, Flavio

if applicable, handling of third-party software providers; and

[11:56 AM] Ceolin, Flavio

what that means ?

[12:05 PM] Gregory A Shue

Good question. Notice that the "manufacturer" does not have to follow a secure SDLC if they are rebranding a fully developed product. I expect the liability gets passed to the OEM, and the OEM must follow a secure SDLC.

 

AFAICT, if the manufacturer uses 3rd part code, then the processes and code of the 3rd party is also part of the assessment. If the 3rd party code is a binary that has already passed the assessment and cannot be impacted by the manufacturer's code (e.g., signed Linux kernel image), then the conformance statement for the binary can be resubmitted by the manufacturer.
```

**Pull request closed without merging**